common/src/main/java/com/lightstep/tracer/shared/PropagatorStack.java

```diff
@@ -0,0 +1,93 @@
+package com.lightstep.tracer.shared;
+
+
+import io.opentracing.propagation.Format;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * PropagatorStack contains a list of {@link Propagator} instances,
+ * while exposing a single {@link Propagator} interface. {@link Propagator}
+ * instances are added through {@link pushPropagator}.
+ *
+ * On {@link extract}, PropagatorStack tries each {@link Propagator}, starting
+ * at the top, stopping with the first {@link io.opentracing.SpanContext} that is
+ * successfully extracted.
+ *
+ * On {@link inject}, PropagatorStack inserts all propagation formats into
+ * the carrier.
+ */
+public final class PropagatorStack<C> implements Propagator<C> {
+    Format<C> format;
+    List<Propagator<C>> propagators;
+
+    /**
+     * Creates a new PropagatorStack associated witht he specified
+     * {@link io.opentracing.propagation.Format}. It is an error
+     * to specify a format that is contained in
+     * {link io.opentracing.propagation.Format#Builtin}.
+     *
+     * The builtin {@link Propagator} for the specified format will be
+     * pushed as the first instance in the stack.
+     *
+     * @param format Instance of {@link io.opentracing.propagation.Format}
+     *               associated with this PropagatorStack.
+     */
+    public PropagatorStack(Format<C> format) {
+        if (format == null) {
+            throw new IllegalArgumentException("format cannot be null");
+        }
+
+        this.format = format;
+        propagators = new LinkedList<Propagator<C>>();
+
+        addBuiltinPropagator();
```

Comment:
I don't believe we should automatically add our default in this manner.

a) User might not want it
b) they may want it, but in a different stack position on extract.